core/client/src/main/java/alluxio/client/block/StreamFactory.java

```diff
@@ -103,13 +103,14 @@ public static BlockInStream createRemoteBlockInStream(FileSystemContext context,
    * @param blockSize the block size
    * @param blockStart the start position of the block in the UFS file
    * @param address the worker network address
+   * @param mountId the id of the mount that this file is mapped to
    * @param options the in stream options
    * @return the input stream
    */
   public static BlockInStream createUfsBlockInStream(FileSystemContext context, String ufsPath,
-      long blockId, long blockSize, long blockStart, WorkerNetAddress address,
+      long blockId, long blockSize, long blockStart, WorkerNetAddress address, long mountId,
```

Comment:
I think overloading the `String` version of the path with the `mountId` is not a good idea, however we could package them together in a `UFSUri`, since most components are currently treating `ufsPath` and `mountId` as a compound key, this wouldn't be hard to do and it clearly shows that `ufsPath` is expected to have a `moundId` since without one the path is no longer useful.